src/solvers/lowering/byte_operators.cpp

```diff
@@ -180,7 +180,46 @@ static exprt unpack_rec(
       max_bytes,
       ns);
   }
-  else if(ns.follow(src.type()).id()==ID_struct)
+  else if(src.type().id() == ID_complex)
+  {
+    const complex_typet &complex_type = to_complex_type(src.type());
+    const typet &subtype = complex_type.subtype();
+
+    auto subtype_bits = pointer_offset_bits(subtype, ns);
+    CHECK_RETURN(subtype_bits.has_value());
+
+    exprt::operandst byte_operands;
+
+    exprt sub_real = unpack_rec(
+      complex_real_exprt(src),
+      little_endian,
+      mp_integer{0},
+      *subtype_bits / 8,
+      ns,
+      true);
+    byte_operands.insert(
+      byte_operands.end(),
+      sub_real.operands().begin(),
+      sub_real.operands().end());
+
+    exprt sub_imag = unpack_rec(
+      complex_imag_exprt(src),
+      little_endian,
+      mp_integer{0},
+      *subtype_bits / 8,
+      ns,
+      true);
+    byte_operands.insert(
+      byte_operands.end(),
+      sub_imag.operands().begin(),
+      sub_imag.operands().end());
+
+    const std::size_t size = byte_operands.size();
+    return array_exprt(
+      std::move(byte_operands),
+      array_typet(unsignedbv_typet(8), from_integer(size, size_type())));
+  }
```

Comment:
The coding standard says "If a method is bigger than 50 lines, break it into parts." so I would recommend putting the code you add into helper functions and call them from here.

Reply:
Done.